pkg/store/kotsstore/version_store.go

```diff
@@ -366,35 +360,6 @@ func (s *KOTSStore) createAppVersion(tx *sql.Tx, appID string, currentSequence *
 	for _, d := range downstreams {
 		// there's a small chance this is not optimal, but no current code path
 		// will support multiple downstreams, so this is cleaner here for now
-		licenseSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "License")
-		if err != nil {
-			return int64(0), errors.Wrap(err, "failed to marshal license spec")
-		}
-		configSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "Config")
-		if err != nil {
-			return int64(0), errors.Wrap(err, "failed to marshal config spec")
-		}
-		configValuesSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "ConfigValues")
-		if err != nil {
-			return int64(0), errors.Wrap(err, "failed to marshal configvalues spec")
-		}
-		identityConfigSpec, err := kotsKinds.Marshal("kots.io", "v1beta1", "IdentityConfig")
-		if err != nil {
-			return int64(0), errors.Wrap(err, "failed to marshal identityconfig spec")
-		}
-
-		configOpts := kotsconfig.ConfigOptions{
```

Comment:
i remember this being used for checking if the config contains any required items that are not set. where's that logic now?

Reply:
i didn't remove any code here.  this is all unused variables.

Reply:
after discussion: this needs to be put back and modified to fix required config items in a separate pr